backend/src/main/kotlin/cz/loono/backend/api/service/ExaminationRecordService.kt

```diff
@@ -34,21 +35,38 @@ class ExaminationRecordService(
     fun cancelExam(examUuid: String, accountUuid: String): ExaminationRecordDto =
         changeState(examUuid, accountUuid, ExaminationStatusDto.CANCELED)
 
-    fun createOrUpdateExam(examinationRecordDto: ExaminationRecordDto, uuid: String): ExaminationRecordDto {
+    fun createOrUpdateExam(examinationRecordDto: ExaminationRecordDto, accountUuid: String): ExaminationRecordDto {
+        validateAccountPrerequisites(examinationRecordDto.type, accountUuid)
         val record = validateUpdateAttempt(examinationRecordDto)
         return examinationRecordRepository.save(
             ExaminationRecord(
                 id = record.id,
                 uuid = record.uuid,
                 type = examinationRecordDto.type,
                 plannedDate = examinationRecordDto.date,
-                account = findAccount(uuid),
+                account = findAccount(accountUuid),
                 firstExam = examinationRecordDto.firstExam ?: true,
                 status = examinationRecordDto.status ?: ExaminationStatusDto.NEW
             )
         ).toExaminationRecordDto()
     }
 
+    private fun validateAccountPrerequisites(type: ExaminationTypeEnumDto, accountUuid: String) {
+        val account = accountRepository.findByUid(accountUuid) ?: throw LoonoBackendException(
+            HttpStatus.NOT_FOUND,
+            "404",
+            "The account not found."
+        )
+        val intervals = preventionService.getExaminationRequests(account).filter { it.examinationType == type }
+        if (intervals.isEmpty()) {
+            throw LoonoBackendException(
+                HttpStatus.BAD_REQUEST,
+                "400",
+                "The account doesn't have rights to create this type of examinations."
+            )
+        }
+    }
```

Comment:
Just a little formatting thing, in Kotlin we can a shorter way starting from 1.6.0
```suggestion
        intervals.ifEmpty {
            throw LoonoBackendException(
                HttpStatus.BAD_REQUEST,
                "400",
                "The account doesn't have rights to create this type of examinations."
            )
        }
    }
```